library/ecp.c

```diff
@@ -381,6 +381,9 @@ static const mbedtls_ecp_curve_info ecp_supported_curves[] =
 #endif
 #if defined(MBEDTLS_ECP_DP_SECP192K1_ENABLED)
     { MBEDTLS_ECP_DP_SECP192K1,    18,     192,    "secp192k1"         },
+#endif
+#if defined(MBEDTLS_ECP_DP_CURVE25519_ENABLED)
+    { MBEDTLS_ECP_DP_CURVE25519,   29,     256,    "x25519"            },
```

Comment:
As far as I know we don't. It is hardcoded in the signature function: it returns an error if you try to sign with an ECDH only curve. In the future we could consider introducing a more elegant design, but for now I think the best solution would be to make the `benchmark` program smarter.

Reply:
This problem isn't unique to the benchmark program. Application code can enumerate the curve list to find curves with the properties it wants, and until now that list only had curves that worked for ECDSA. It may be unavoidable that some existing code breaks, but at least we should provide a transition path, something like “if you were enumerating curves and you want a curve that's good for ECDSA, you now need add an additional check `mbedtls_ecdsa_can_do( grp )`”.

Reply:
Added!